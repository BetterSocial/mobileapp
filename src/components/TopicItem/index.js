/* eslint-disable arrow-body-style */
import * as React from 'react';
import Icon from 'react-native-vector-icons/Fontisto';
import {Pressable, StyleSheet, Text, TouchableOpacity, View} from 'react-native';

import {fonts} from '../../utils/fonts';
import {COLORS} from '../../utils/theme';

const TopicItem = ({label, removeTopic, style, onTopicPress = () => {}}) => {
  return (
    <Pressable testID="topicPress" onPress={onTopicPress}>
      <View style={[styles.containerTag, style]}>
        <Text style={styles.tag}>{label}</Text>
        <TouchableOpacity
          testID="removeTopic"
          onPress={() => removeTopic(label)}
          style={styles.btn}>
          <Icon name="close" size={13.33} allowFontScaling={false} />
        </TouchableOpacity>
      </View>
    </Pressable>
  );
};

export default TopicItem;

const styles = StyleSheet.create({
  containerTag: {
    backgroundColor: COLORS.almostBlack,
    borderRadius: 14,
    paddingLeft: 16,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginRight: 12
  },
  tag: {
    fontFamily: fonts.inter[400],
    fontSize: 12
  },
  btn: {
    marginLeft: 7,
    paddingVertical: 10,
    paddingLeft: 4,
    paddingRight: 10
  }
});
