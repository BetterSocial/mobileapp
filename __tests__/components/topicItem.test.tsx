import * as React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import TopicItem from '../../src/components/TopicItem';

configure({adapter: new Adapter()});
describe('component TopicItem', () => {
  it('TopicItem snapshot', () => {
    const component = shallow(
      <TopicItem label="Test" removeTopic={() => {}} style={{}} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('TopicItem renders correctly', () => {
    const tree = renderer
      .create(<TopicItem label="Test" removeTopic={() => {}} style={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
