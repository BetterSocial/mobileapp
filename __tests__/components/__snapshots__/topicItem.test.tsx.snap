// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component TopicItem TopicItem renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 14,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginRight": 12,
        "paddingLeft": 16,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Inter-Regular",
        "fontSize": 12,
      }
    }
  >
    Test
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 7,
        "opacity": 1,
        "padding": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="close"
      size={13.33}
    />
  </View>
</View>
`;

exports[`component TopicItem TopicItem snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 14,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginRight": 12,
        "paddingLeft": 16,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Inter-Regular",
        "fontSize": 12,
      }
    }
  >
    Test
  </Text>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "marginLeft": 7,
        "padding": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="close"
      size={13.33}
    />
  </ForwardRef>
</View>
`;
